<!-- <launch>
    
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="cmd_vel_topic" default="/cmd_vel" />
    <arg name="odom_topic" default="odom" />
    <arg name="move_forward_only" default="false"/>
  
   
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
      <rosparam file="$(find super_robot_package)/config/costmap_common_params_waffle.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find super_robot_package)/config/costmap_common_params_waffle.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find super_robot_package)/config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find super_robot_package)/config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find super_robot_package)/config/move_base_params.yaml" command="load" />
      <rosparam file="$(find super_robot_package)/config/dwa_local_planner_params_$(arg model).yaml" command="load" />
      <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
      <remap from="odom" to="$(arg odom_topic)"/>
      <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  
    </node>
  </launch> -->
  
  <?xml version="1.0"?>
<launch>
  <arg name="odom_frame"   default="odom"/>
  <arg name="base_frame"   default="base_footprint"/>
  <arg name="global_frame" default="map"/>
  <arg name="cmd_topic"    default="cmd_vel" />
  <arg name="odom_topic"   default="odom" />
  <arg name="laser_topic"  default="scan" />
  <arg name="config_dir"   default="config" />
  <arg name="new_dir" default="new"/>
  <arg name="wp_global_planner" default="false" doc="[true - wp_global_planner, false - global_planner]"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find wr8_software)/$(arg config_dir)/$(arg new_dir)/costmap_common_params_waffle.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find wr8_software)/$(arg config_dir)/$(arg new_dir)/costmap_common_params_waffle.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find wr8_software)/$(arg config_dir)/$(arg new_dir)/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find wr8_software)/$(arg config_dir)/$(arg new_dir)/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find wr8_software)/$(arg config_dir)/$(arg new_dir)/move_base_params.yaml" command="load" />


    <rosparam file="$(find wr8_software)/$(arg config_dir)/wp_global_planner_params.yaml" command="load" if="$(arg wp_global_planner)"/>
    <rosparam file="$(find wr8_software)/$(arg config_dir)/global_planner_params.yaml" command="load" unless="$(arg wp_global_planner)"/>  

    <rosparam file="$(find wr8_software)/$(arg config_dir)/teb_local_planner_params.yaml" command="load" />
    
    <param name="global_costmap/global_frame" value="$(arg global_frame)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame)"/>
    <param name="local_costmap/global_frame" value="$(arg global_frame)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame)"/>

    <param name="base_global_planner" value="wp_global_planner/WPGlobalPlanner" if="$(arg wp_global_planner)"/>
    <param name="base_global_planner" value="global_planner/GlobalPlanner" unless="$(arg wp_global_planner)"/>

    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"/>

    <remap from="cmd_vel" to="$(arg cmd_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>
